File: Assignment41_3.c
#include <stdio.h>

///////////////////////////////////////////////////////////////////////////////
// Function Name : Strlen
// Description   : Recursively count characters in string
// Input         : String
// Output        : Count of characters
// Author        : Raj Samir Jadhav
// Date          : 06/12/2025
///////////////////////////////////////////////////////////////////////////////


int Strlen(char *str)
{
    static int iCnt = 0;

    if(*str != '\0')       // Base condition
    {
        iCnt++;
        Strlen(str + 1);   // Recursive call with next character
    }

    return iCnt;
}

int main()
{
    int iRet = 0;
    char arr[20];

    printf("Enter string : ");
    scanf("%s", arr);

    iRet = Strlen(arr);

    printf("%d\n", iRet);

    return 0;
}

File: Assignment41_3.java
import java.util.Scanner;

///////////////////////////////////////////////////////////////////////////////
// Function Name : Strlen
// Description   : Recursively count characters in string
// Input         : String
// Output        : Count of characters
// Author        : Raj Samir Jadhav
// Date          : 06/12/2025
///////////////////////////////////////////////////////////////////////////////
class Pattern
{
    int iCnt = 0;

    int Strlen(String str, int index)
    {
        if(index < str.length())
        {
            iCnt++;
            Strlen(str, index + 1);  // Recursive step
        }
        return iCnt;
    }
}

class Assignment41_3
{
    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        System.out.print("Enter string : ");
        String input = sobj.nextLine();

        Pattern pobj = new Pattern();
        int result = pobj.Strlen(input, 0);

        System.out.println(result);

        sobj.close();
    }
}
